rs/sns/governance/canister/governance.did

```diff
@@ -21,6 +21,135 @@ type Action = variant {
   ExecuteGenericNervousSystemFunction : ExecuteGenericNervousSystemFunction;
   ManageLedgerParameters : ManageLedgerParameters;
   Motion : Motion;
+  RegisterExtension : RegisterExtension;
+  DeregisterExtension : DeregisterExtension;
+  ExecuteExtensionFunction : ExecuteExtensionFunction;
+};
+
+type ExistingCanister = record {
+  canister_id : opt principal;
+  wasm_module_hash : opt blob;
+};
+
+type NewCanister = record {
+  chunked_canister_wasm : opt ChunkedCanisterWasm;
+};
+
+type Installation = variant {
+  existing_canister : ExistingCanister;
+  new_canister : NewCanister;
```

Comment:
I think extensions should always be NewCanister, b/c you want to make the chain of custody of the internal state of the canister easy to track.

If I install a canister somewhere, I could maybe fiddle with the state and make it do surprising things later, if I had enough knowledge and there were some exploitable edge cases, but the wasm hash would still be valid.

Reply:
> In particular, the client might want to know the ID of that canister before registering it with the SNS.
What use case do you have in mind?

I agree it's possible, but it's also a lot more work than just clicking a button and knowing the SNS you trust installed a canister in a trusted way that's from trusted source... Things are more secure when they're easier to check.

Reply:
Okay, agreed! Thanks for making the arguments very clear.